src/database/database-mariadb.cpp

```diff
@@ -0,0 +1,1407 @@
+	try {
+
+		stmtDeleteBlock->setInt(1, pos.X);
+		stmtDeleteBlock->setInt(2, pos.Y);
+		stmtDeleteBlock->setInt(3, pos.Z);
+		stmtDeleteBlock->executeUpdate();
+		stmtDeleteBlock->clearParameters();
+
+		std::int32_t rows_affected = stmtDeleteBlock->getUpdateCount();
+
+		return (rows_affected > 0);
+
+	} catch (sql::SQLException &e) {
+		errorstream << "[MariaDB] Error: " << + e.what() << std::endl;
+		throw DatabaseException("MariaDB Error");
+	}
+}
+
+
+void MapDatabaseMariaDB::listAllLoadableBlocks(std::vector<v3s16> &dst) {
+
+	checkConnection();
+
+	try {
+
+		std::unique_ptr<sql::ResultSet> resBlocks(stmtListAllLoadableBlocks->executeQuery());
+
+		while (resBlocks->next()) {
+		
+			v3s16 row(resBlocks->getInt("x"), resBlocks->getInt("y"), resBlocks->getInt("z"));
+			dst.push_back(row);
+		
+		}
+		
+		resBlocks->close();
+
+	} catch (sql::SQLException &e) {
+		errorstream << "[MariaDB] Error: " << + e.what() << std::endl;
+		throw DatabaseException("MariaDB Error");
+	}
+}
+
+
+void MapDatabaseMariaDB::loadBlock(const v3s16 &pos, std::string *block) {
+
+	checkConnection();
+
+	try {
+
+		stmtLoadBlock->setInt(1, htonl(pos.X));
+		stmtLoadBlock->setInt(2, htonl(pos.Y));
+		stmtLoadBlock->setInt(3, htonl(pos.Z));
+		
+		std::unique_ptr<sql::ResultSet> resBlock(stmtLoadBlock->executeQuery());
+
+		if (resBlock->next()) {
+
+			std::stringstream in;
+			in << resBlock->getBlob("data")->rdbuf();
+			std::string data = std::move(in).str();
+			*block = data;
+
+		} else {
+
+			block->clear();
+
+		}
+
+		resBlock->close();
+		stmtLoadBlock->clearParameters();
+
+	} catch (sql::SQLException &e) {
+		errorstream << "[MariaDB] Error: " << + e.what() << std::endl;
+		throw DatabaseException("MariaDB Error");
+	}
+}
+
+
+bool MapDatabaseMariaDB::saveBlock(const v3s16 &pos, const std::string &data) {
+
+	checkConnection();
+
+	try {
+
+		size_t max_size = 4294967295;
+		if (data.size() > max_size) {
+
+			errorstream << "[MariaDB] Error: refusing to save block because data would be truncated: " << data.size() << " > " << max_size << std::endl;
+			return false;
+
+		}
+
+		std::istringstream _data(data);
+
+		stmtSaveBlock->setInt(1, htonl(pos.X));
+		stmtSaveBlock->setInt(2, htonl(pos.Y));
+		stmtSaveBlock->setInt(3, htonl(pos.Z));
+		stmtSaveBlock->setBlob(4, &_data);
+		stmtSaveBlock->executeUpdate();
+		stmtSaveBlock->clearParameters();
+
+		std::int32_t rows_affected = stmtSaveBlock->getUpdateCount();
+
+		return (rows_affected > 0);
+
+	} catch (sql::SQLException &e) {
+		errorstream << "[MariaDB] Error: " << + e.what() << std::endl;
+		throw DatabaseException("MariaDB Error");
+	}
+}
+
+
+/*
+ *
+ * PlayerDatabaseMariaDB -- The MariaDB Player Database Backend
+ * 
+ */
+
+
+PlayerDatabaseMariaDB::PlayerDatabaseMariaDB(const std::string &connect_string)
+: Database_MariaDB(connect_string, "_player"), PlayerDatabase() {
+	
+	// connect to database
+	connect();
+
+	// create tables if they do n
+	createTable("player",
+		"name VARCHAR(60),"
```

Comment:
i'm not sure about the VARCHAR(60), i think you should use TEXT like on PG to prevent bad silent truncates

Reply:
Player username absolutely does not need to be unlimited, I believe it's already limited to something like 18-20 characters

Reply:
@rubenwardy do you know that mysql will just truncate field silently and it's  not a constraint ?

Reply:
If that's the cast, then it should be enforced by the server at an earlier point

Reply:
it should, varchar is just a text with limitation on storage size, there is zero interest db side. in terms of index it's same as text, then no interest too :)

Reply:
The max player name size is 20, as defined in player.h. Attempts to connect with a name longer than this is rejected.

#define PLAYERNAME_SIZE 20

Reply:
> it should, varchar is just a text with limitation on storage size, there is zero interest db side. in terms of index it's same as text, then no interest too :)

Please rephrase because I have no idea what you mean

Reply:
I'm guessing you're referring to the fact that the performance of TEXT and VARCHAR are the same in PostgreSQL:

> Both TEXT and VARCHAR have the upper limit at 1 Gb, and there is no performance difference among them (according to the PostgreSQL documentation).

This is not the case for MariaDB, where VARCHAR is faster. Additionally, using VARCHAR with postgresql still has the benefit of limiting the size of stored data

Reply:
Fixed in latest commit. Now importing and using PLAYERNAME_SIZE from player.h.